services/media.py

```
@@ -0,0 +1,67 @@
+def probe_duration(path: str) -> Optional[float]:
+    """
+    Probe video duration using ffprobe.
+    
+    Args:
+        path: Path to video file
+        
+    Returns:
+        Duration in seconds, or None if ffprobe fails or is unavailable
+    """
+    try:
+        result = subprocess.run([
```

Ruff (0.12.2)

`subprocess` call: check for execution of untrusted input

(S603)